src/Main.hs

```diff
@@ -214,6 +217,19 @@ readDerivation path = do
         _ -> do
             fail ("Could not parse a derivation from this file: " ++ string)
 
+-- | Read and parse a derivation from a store path that can be a derivation
+-- (.drv) or a realized path, in which case the corresponding derivation is
+-- queried.
+readInput :: FilePath -> Diff (Derivation FilePath Text)
+readInput path =
+    let string = path in
+    if List.isSuffixOf ".drv" string
+    then readDerivation path
+    else do
+        result <- liftIO (Process.readProcess "nix-store" [ "-qd", string ] [])
```

Comment:
For non-interactive commands, I try to use the long form of option names so that the code is more self-documenting:

```suggestion
        result <- liftIO (Process.readProcess "nix-store" [ "--query", "--deriver", string ] [])
```

Reply:
Good idea.